### ISSUE TITLE
All NTAG215 reads of a blank card appear with 2 pages locked

### ISSUE THREAD

**Issue by johng3587** [flipperdevices/flipperzero-firmware#3281]
### Describe the bug.

I noticed that recently my flipper started saying all my NTAG215 cards are password protected, and can only read 133/135 pages. This didn't happen before whatever changed. The 
NFC tools app on my phone says the card is not password protected. I even tried a fresh card, never used, still says password protected. Recently, I had been messing around with password protecting NTAG215 cards on my phone, and then unlocking them with the flipper, this could have been the start of the problem, but I'm not sure.

![Screenshot 2023-12-09 112347](https://github.com/flipperdevices/flipperzero-firmware/assets/57050681/583fc0c4-b24e-49e7-bea5-a3edc7da3278)
![Screenshot 2023-12-09 112400](https://github.com/flipperdevices/flipperzero-firmware/assets/57050681/189e0250-cff6-47e7-9a5a-c4d9c175c13c)

### Reproduction

1. Get blank NTAG215 card
2. Read card with regular NFC app on flipper
3. Flipper should say there are some password-protected pages at this point.
4. Unlock with password "FFFFFFFF"
5. Read card again
6. Flipper should say all pages unlocked.

### Logs

```Text
53511 [I][Loader] Loading /ext/apps/NFC/nfc.fap
53648 [I][Elf] Total size of loaded sections: 53013
53650 [I][Loader] Loaded in 139ms
53654 [I][AnimationManager] Unload animation 'L2_Soldering_128x64'
54736 [D][NfcScanner] Found 5 base protocols
54744 [D][DolphinState] icounter 427, butthurt 0
54796 [D][Nfc] FWT Timeout
54821 [D][Nfc] FWT Timeout
54873 [D][Nfc] FWT Timeout
54901 [D][Nfc] FWT Timeout
54916 [D][NfcScanner] Found 4 children
54971 [D][Nfc] FWT Timeout
55009 [D][Nfc] FWT Timeout
55043 [E][Iso14443_4aPoller] Failed to parse ATS response
55045 [D][Iso14443_4aPoller] Failed to read ATS
55047 [D][Nfc] FWT Timeout
55063 [I][NfcScanner] Detected 1 protocols
55193 [D][MfUltralightPoller] Read version success
55195 [D][MfUltralightPoller] NTAG215 detected. Total pages: 135
55197 [D][MfUltralightPoller] Reading signature
55205 [D][MfUltralightPoller] Read page 0 success
55207 [D][MfUltralightPoller] Read page 1 success
55209 [D][MfUltralightPoller] Read page 2 success
55211 [D][MfUltralightPoller] Read page 3 success
55215 [D][MfUltralightPoller] Read page 4 success
55217 [D][MfUltralightPoller] Read page 5 success
55219 [D][MfUltralightPoller] Read page 6 success
55221 [D][MfUltralightPoller] Read page 7 success
55226 [D][MfUltralightPoller] Read page 8 success
55228 [D][MfUltralightPoller] Read page 9 success
55231 [D][MfUltralightPoller] Read page 10 success
55233 [D][MfUltralightPoller] Read page 11 success
55239 [D][MfUltralightPoller] Read page 12 success
55241 [D][MfUltralightPoller] Read page 13 success
55243 [D][MfUltralightPoller] Read page 14 success
55246 [D][MfUltralightPoller] Read page 15 success
55251 [D][MfUltralightPoller] Read page 16 success
55253 [D][MfUltralightPoller] Read page 17 success
55255 [D][MfUltralightPoller] Read page 18 success
55257 [D][MfUltralightPoller] Read page 19 success
55262 [D][MfUltralightPoller] Read page 20 success
55264 [D][MfUltralightPoller] Read page 21 success
55266 [D][MfUltralightPoller] Read page 22 success
55268 [D][MfUltralightPoller] Read page 23 success
55272 [D][MfUltralightPoller] Read page 24 success
55274 [D][MfUltralightPoller] Read page 25 success
55276 [D][MfUltralightPoller] Read page 26 success
55278 [D][MfUltralightPoller] Read page 27 success
55282 [D][MfUltralightPoller] Read page 28 success
55284 [D][MfUltralightPoller] Read page 29 success
55286 [D][MfUltralightPoller] Read page 30 success
55288 [D][MfUltralightPoller] Read page 31 success
55292 [D][MfUltralightPoller] Read page 32 success
55294 [D][MfUltralightPoller] Read page 33 success
55296 [D][MfUltralightPoller] Read page 34 success
55298 [D][MfUltralightPoller] Read page 35 success
55302 [D][MfUltralightPoller] Read page 36 success
55304 [D][MfUltralightPoller] Read page 37 success
55306 [D][MfUltralightPoller] Read page 38 success
55308 [D][MfUltralightPoller] Read page 39 success
55312 [D][MfUltralightPoller] Read page 40 success
55314 [D][MfUltralightPoller] Read page 41 success
55316 [D][MfUltralightPoller] Read page 42 success
55318 [D][MfUltralightPoller] Read page 43 success
55322 [D][MfUltralightPoller] Read page 44 success
55324 [D][MfUltralightPoller] Read page 45 success
55326 [D][MfUltralightPoller] Read page 46 success
55328 [D][MfUltralightPoller] Read page 47 success
55332 [D][MfUltralightPoller] Read page 48 success
55334 [D][MfUltralightPoller] Read page 49 success
55336 [D][MfUltralightPoller] Read page 50 success
55338 [D][MfUltralightPoller] Read page 51 success
55342 [D][MfUltralightPoller] Read page 52 success
55344 [D][MfUltralightPoller] Read page 53 success
55346 [D][MfUltralightPoller] Read page 54 success
55348 [D][MfUltralightPoller] Read page 55 success
55352 [D][MfUltralightPoller] Read page 56 success
55354 [D][MfUltralightPoller] Read page 57 success
55356 [D][MfUltralightPoller] Read page 58 success
55358 [D][MfUltralightPoller] Read page 59 success
55362 [D][MfUltralightPoller] Read page 60 success
55364 [D][MfUltralightPoller] Read page 61 success
55366 [D][MfUltralightPoller] Read page 62 success
55368 [D][MfUltralightPoller] Read page 63 success
55372 [D][MfUltralightPoller] Read page 64 success
55375 [D][MfUltralightPoller] Read page 65 success
55378 [D][MfUltralightPoller] Read page 66 success
55380 [D][MfUltralightPoller] Read page 67 success
55384 [D][MfUltralightPoller] Read page 68 success
55387 [D][MfUltralightPoller] Read page 69 success
55390 [D][MfUltralightPoller] Read page 70 success
55393 [D][MfUltralightPoller] Read page 71 success
55398 [D][MfUltralightPoller] Read page 72 success
55401 [D][MfUltralightPoller] Read page 73 success
55403 [D][MfUltralightPoller] Read page 74 success
55406 [D][MfUltralightPoller] Read page 75 success
55412 [D][MfUltralightPoller] Read page 76 success
55414 [D][MfUltralightPoller] Read page 77 success
55417 [D][MfUltralightPoller] Read page 78 success
55420 [D][MfUltralightPoller] Read page 79 success
55425 [D][MfUltralightPoller] Read page 80 success
55427 [D][MfUltralightPoller] Read page 81 success
55429 [D][MfUltralightPoller] Read page 82 success
55431 [D][MfUltralightPoller] Read page 83 success
55436 [D][MfUltralightPoller] Read page 84 success
55438 [D][MfUltralightPoller] Read page 85 success
55440 [D][MfUltralightPoller] Read page 86 success
55442 [D][MfUltralightPoller] Read page 87 success
55446 [D][MfUltralightPoller] Read page 88 success
55448 [D][MfUltralightPoller] Read page 89 success
55450 [D][MfUltralightPoller] Read page 90 success
55452 [D][MfUltralightPoller] Read page 91 success
55456 [D][MfUltralightPoller] Read page 92 success
55458 [D][MfUltralightPoller] Read page 93 success
55460 [D][MfUltralightPoller] Read page 94 success
55462 [D][MfUltralightPoller] Read page 95 success
55466 [D][MfUltralightPoller] Read page 96 success
55468 [D][MfUltralightPoller] Read page 97 success
55470 [D][MfUltralightPoller] Read page 98 success
55472 [D][MfUltralightPoller] Read page 99 success
55476 [D][MfUltralightPoller] Read page 100 success
55478 [D][MfUltralightPoller] Read page 101 success
55480 [D][MfUltralightPoller] Read page 102 success
55482 [D][MfUltralightPoller] Read page 103 success
55486 [D][MfUltralightPoller] Read page 104 success
55488 [D][MfUltralightPoller] Read page 105 success
55490 [D][MfUltralightPoller] Read page 106 success
55492 [D][MfUltralightPoller] Read page 107 success
55496 [D][MfUltralightPoller] Read page 108 success
55499 [D][MfUltralightPoller] Read page 109 success
55502 [D][MfUltralightPoller] Read page 110 success
55505 [D][MfUltralightPoller] Read page 111 success
55511 [D][MfUltralightPoller] Read page 112 success
55513 [D][MfUltralightPoller] Read page 113 success
55515 [D][MfUltralightPoller] Read page 114 success
55517 [D][MfUltralightPoller] Read page 115 success
55521 [D][MfUltralightPoller] Read page 116 success
55523 [D][MfUltralightPoller] Read page 117 success
55525 [D][MfUltralightPoller] Read page 118 success
55527 [D][MfUltralightPoller] Read page 119 success
55531 [D][MfUltralightPoller] Read page 120 success
55533 [D][MfUltralightPoller] Read page 121 success
55535 [D][MfUltralightPoller] Read page 122 success
55537 [D][MfUltralightPoller] Read page 123 success
55541 [D][MfUltralightPoller] Read page 124 success
55543 [D][MfUltralightPoller] Read page 125 success
55545 [D][MfUltralightPoller] Read page 126 success
55547 [D][MfUltralightPoller] Read page 127 success
55552 [D][MfUltralightPoller] Read page 128 success
55554 [D][MfUltralightPoller] Read page 129 success
55556 [D][MfUltralightPoller] Read page 130 success
55558 [D][MfUltralightPoller] Read page 131 success
55562 [D][MfUltralightPoller] Read page 132 success
55564 [D][MfUltralightPoller] Read page 133 success
55566 [D][MfUltralightPoller] Read page 134 success
55568 [D][MfUltralightPoller] Skip reading counters
55570 [D][MfUltralightPoller] Reading tearing flag 2
55573 [D][MfUltralightPoller] No limits in authentication. Trying default password
55580 [D][Nfc] FWT Timeout
55582 [D][MfUltralightPoller] Read success
55585 [D][Nfc] FWT Timeout
55647 [I][Elf] Total size of loaded sections: 888
55650 [D][Fap] Library for NfcSupportedCardPlugin, API v. 1 loaded
55690 [I][Elf] Total size of loaded sections: 420
55693 [D][Fap] Library for NfcSupportedCardPlugin, API v. 1 loaded
55695 [I][AllInOne] Pass not verified
55744 [I][Elf] Total size of loaded sections: 1244
55746 [D][Fap] Library for NfcSupportedCardPlugin, API v. 1 loaded
55790 [I][Elf] Total size of loaded sections: 1324
55792 [D][Fap] Library for NfcSupportedCardPlugin, API v. 1 loaded
55837 [I][Elf] Total size of loaded sections: 1724
55840 [D][Fap] Library for NfcSupportedCardPlugin, API v. 1 loaded
55883 [I][Elf] Total size of loaded sections: 1768
55885 [D][Fap] Library for NfcSupportedCardPlugin, API v. 1 loaded
55928 [I][Elf] Total size of loaded sections: 1464
55931 [D][Fap] Library for NfcSupportedCardPlugin, API v. 1 loaded
55937 [D][DolphinState] icounter 427, butthurt 0
60777 [D][BtGap] set_non_discoverable success
65030 [D][DolphinState] icounter 427, butthurt 0
65039 [D][MfUltralightPoller] Read version success
65041 [D][MfUltralightPoller] NTAG215 detected. Total pages: 135
65043 [D][MfUltralightPoller] Reading signature
65051 [D][MfUltralightPoller] Trying to authenticate with password FFFFFFFF
65055 [D][MfUltralightPoller] Auth success
65059 [D][MfUltralightPoller] Read page 0 success
65061 [D][MfUltralightPoller] Read page 1 success
65063 [D][MfUltralightPoller] Read page 2 success
65065 [D][MfUltralightPoller] Read page 3 success
65071 [D][MfUltralightPoller] Read page 4 success
65073 [D][MfUltralightPoller] Read page 5 success
65075 [D][MfUltralightPoller] Read page 6 success
65078 [D][MfUltralightPoller] Read page 7 success
65082 [D][MfUltralightPoller] Read page 8 success
65085 [D][MfUltralightPoller] Read page 9 success
65087 [D][MfUltralightPoller] Read page 10 success
65089 [D][MfUltralightPoller] Read page 11 success
65093 [D][MfUltralightPoller] Read page 12 success
65096 [D][MfUltralightPoller] Read page 13 success
65099 [D][MfUltralightPoller] Read page 14 success
65101 [D][MfUltralightPoller] Read page 15 success
65105 [D][MfUltralightPoller] Read page 16 success
65108 [D][MfUltralightPoller] Read page 17 success
65111 [D][MfUltralightPoller] Read page 18 success
65114 [D][MfUltralightPoller] Read page 19 success
65119 [D][MfUltralightPoller] Read page 20 success
65122 [D][MfUltralightPoller] Read page 21 success
65124 [D][MfUltralightPoller] Read page 22 success
65127 [D][MfUltralightPoller] Read page 23 success
65133 [D][MfUltralightPoller] Read page 24 success
65135 [D][MfUltralightPoller] Read page 25 success
65138 [D][MfUltralightPoller] Read page 26 success
65140 [D][MfUltralightPoller] Read page 27 success
65146 [D][MfUltralightPoller] Read page 28 success
65148 [D][MfUltralightPoller] Read page 29 success
65150 [D][MfUltralightPoller] Read page 30 success
65153 [D][MfUltralightPoller] Read page 31 success
65158 [D][MfUltralightPoller] Read page 32 success
65160 [D][MfUltralightPoller] Read page 33 success
65162 [D][MfUltralightPoller] Read page 34 success
65164 [D][MfUltralightPoller] Read page 35 success
65168 [D][MfUltralightPoller] Read page 36 success
65170 [D][MfUltralightPoller] Read page 37 success
65172 [D][MfUltralightPoller] Read page 38 success
65174 [D][MfUltralightPoller] Read page 39 success
65178 [D][MfUltralightPoller] Read page 40 success
65180 [D][MfUltralightPoller] Read page 41 success
65182 [D][MfUltralightPoller] Read page 42 success
65184 [D][MfUltralightPoller] Read page 43 success
65188 [D][MfUltralightPoller] Read page 44 success
65190 [D][MfUltralightPoller] Read page 45 success
65192 [D][MfUltralightPoller] Read page 46 success
65194 [D][MfUltralightPoller] Read page 47 success
65198 [D][MfUltralightPoller] Read page 48 success
65200 [D][MfUltralightPoller] Read page 49 success
65202 [D][MfUltralightPoller] Read page 50 success
65204 [D][MfUltralightPoller] Read page 51 success
65208 [D][MfUltralightPoller] Read page 52 success
65210 [D][MfUltralightPoller] Read page 53 success
65212 [D][MfUltralightPoller] Read page 54 success
65214 [D][MfUltralightPoller] Read page 55 success
65218 [D][MfUltralightPoller] Read page 56 success
65220 [D][MfUltralightPoller] Read page 57 success
65222 [D][MfUltralightPoller] Read page 58 success
65224 [D][MfUltralightPoller] Read page 59 success
65228 [D][MfUltralightPoller] Read page 60 success
65230 [D][MfUltralightPoller] Read page 61 success
65232 [D][MfUltralightPoller] Read page 62 success
65234 [D][MfUltralightPoller] Read page 63 success
65238 [D][MfUltralightPoller] Read page 64 success
65240 [D][MfUltralightPoller] Read page 65 success
65242 [D][MfUltralightPoller] Read page 66 success
65244 [D][MfUltralightPoller] Read page 67 success
65248 [D][MfUltralightPoller] Read page 68 success
65250 [D][MfUltralightPoller] Read page 69 success
65252 [D][MfUltralightPoller] Read page 70 success
65254 [D][MfUltralightPoller] Read page 71 success
65258 [D][MfUltralightPoller] Read page 72 success
65260 [D][MfUltralightPoller] Read page 73 success
65262 [D][MfUltralightPoller] Read page 74 success
65264 [D][MfUltralightPoller] Read page 75 success
65268 [D][MfUltralightPoller] Read page 76 success
65270 [D][MfUltralightPoller] Read page 77 success
65272 [D][MfUltralightPoller] Read page 78 success
65274 [D][MfUltralightPoller] Read page 79 success
65278 [D][MfUltralightPoller] Read page 80 success
65280 [D][MfUltralightPoller] Read page 81 success
65282 [D][MfUltralightPoller] Read page 82 success
65284 [D][MfUltralightPoller] Read page 83 success
65288 [D][MfUltralightPoller] Read page 84 success
65291 [D][MfUltralightPoller] Read page 85 success
65293 [D][MfUltralightPoller] Read page 86 success
65295 [D][MfUltralightPoller] Read page 87 success
65299 [D][MfUltralightPoller] Read page 88 success
65302 [D][MfUltralightPoller] Read page 89 success
65305 [D][MfUltralightPoller] Read page 90 success
65307 [D][MfUltralightPoller] Read page 91 success
65312 [D][MfUltralightPoller] Read page 92 success
65314 [D][MfUltralightPoller] Read page 93 success
65317 [D][MfUltralightPoller] Read page 94 success
65320 [D][MfUltralightPoller] Read page 95 success
65326 [D][MfUltralightPoller] Read page 96 success
65328 [D][MfUltralightPoller] Read page 97 success
65331 [D][MfUltralightPoller] Read page 98 success
65333 [D][MfUltralightPoller] Read page 99 success
65339 [D][MfUltralightPoller] Read page 100 success
65341 [D][MfUltralightPoller] Read page 101 success
65343 [D][MfUltralightPoller] Read page 102 success
65345 [D][MfUltralightPoller] Read page 103 success
65349 [D][MfUltralightPoller] Read page 104 success
65351 [D][MfUltralightPoller] Read page 105 success
65353 [D][MfUltralightPoller] Read page 106 success
65355 [D][MfUltralightPoller] Read page 107 success
65359 [D][MfUltralightPoller] Read page 108 success
65361 [D][MfUltralightPoller] Read page 109 success
65363 [D][MfUltralightPoller] Read page 110 success
65365 [D][MfUltralightPoller] Read page 111 success
65369 [D][MfUltralightPoller] Read page 112 success
65371 [D][MfUltralightPoller] Read page 113 success
65373 [D][MfUltralightPoller] Read page 114 success
65375 [D][MfUltralightPoller] Read page 115 success
65380 [D][MfUltralightPoller] Read page 116 success
65383 [D][MfUltralightPoller] Read page 117 success
65386 [D][MfUltralightPoller] Read page 118 success
65389 [D][MfUltralightPoller] Read page 119 success
65394 [D][MfUltralightPoller] Read page 120 success
65396 [D][MfUltralightPoller] Read page 121 success
65398 [D][MfUltralightPoller] Read page 122 success
65400 [D][MfUltralightPoller] Read page 123 success
65404 [D][MfUltralightPoller] Read page 124 success
65406 [D][MfUltralightPoller] Read page 125 success
65408 [D][MfUltralightPoller] Read page 126 success
65410 [D][MfUltralightPoller] Read page 127 success
65414 [D][MfUltralightPoller] Read page 128 success
65416 [D][MfUltralightPoller] Read page 129 success
65418 [D][MfUltralightPoller] Read page 130 success
65420 [D][MfUltralightPoller] Read page 131 success
65425 [D][MfUltralightPoller] Read page 132 success
65428 [D][MfUltralightPoller] Read page 133 success
65431 [D][MfUltralightPoller] Read page 134 success
65434 [D][MfUltralightPoller] Skip reading counters
65437 [D][MfUltralightPoller] Reading tearing flag 2
65440 [D][MfUltralightPoller] Read success
65443 [D][Nfc] FWT Timeout
65504 [I][Elf] Total size of loaded sections: 888
65508 [D][Fap] Library for NfcSupportedCardPlugin, API v. 1 loaded
65547 [I][Elf] Total size of loaded sections: 420
65549 [D][Fap] Library for NfcSupportedCardPlugin, API v. 1 loaded
65552 [I][AllInOne] Pass not verified
65599 [I][Elf] Total size of loaded sections: 1244
65603 [D][Fap] Library for NfcSupportedCardPlugin, API v. 1 loaded
65644 [I][Elf] Total size of loaded sections: 1324
65647 [D][Fap] Library for NfcSupportedCardPlugin, API v. 1 loaded
65699 [I][Elf] Total size of loaded sections: 1724
65703 [D][Fap] Library for NfcSupportedCardPlugin, API v. 1 loaded
65746 [I][Elf] Total size of loaded sections: 1768
65749 [D][Fap] Library for NfcSupportedCardPlugin, API v. 1 loaded
65795 [I][Elf] Total size of loaded sections: 1464
65797 [D][Fap] Library for NfcSupportedCardPlugin, API v. 1 loaded
65803 [D][DolphinState] icounter 427, butthurt 0
74730 [D][ViewDispatcher] View changed while key press 2000B838 -> 2000B158. Sending key: Back, type: Release, sequence: 00000100 to previous view port
75840 [I][Loader] App returned: 0
75848 [I][Loader] Application stopped. Free heap: 143272
76200 [I][AnimationManager] Select 'L2_Soldering_128x64' animation
76204 [I][AnimationManager] Load animation 'L2_Soldering_128x64'
76208 [I][SavedStruct] Loading "/int/.desktop.settings"
76233 [D][ViewDispatcher] View changed while key press 20009EA0 -> 00000000. Sending key: Back, type: Release, sequence: 0000010C to previous view port
```

### Anything else?

I notice in the debug logs, it says "Read page 134 success" on line 55566, and "Read success" on line 55582, this was before I tried the password. Flipper says 133/135 pages read, but the logs say read successful. If page 134 was successful, why does flipper say only 133/135 could be read? Logs also say it tries the default password, before saying read success.

**Comment by johng3587** [comment]
![image](https://github.com/flipperdevices/flipperzero-firmware/assets/57050681/ac0f4a3c-4eeb-47b5-9b6b-473901695115)
```
Current log level: debug
Use <log ?> to list available log levels
Press CTRL+C to stop...
26340679 [D][WMFU] Card searching...
26340684 [D][DolphinState] icounter 451, butthurt 0
26340699 [D][Nfc] FWT Timeout
26340801 [D][Nfc] FWT Timeout
26340911 [D][MfUltralightPoller] Read version success
26340913 [D][MfUltralightPoller] NTAG215 detected. Total pages: 135
26340916 [D][MfUltralightPoller] Reading signature
26340923 [D][MfUltralightPoller] Read page 0 success
26340926 [D][MfUltralightPoller] Read page 1 success
26340930 [D][MfUltralightPoller] Read page 2 success
[...]
26341284 [D][MfUltralightPoller] Read page 133 success
26341287 [D][MfUltralightPoller] Read page 134 success
26341289 [D][MfUltralightPoller] Skip reading counters
26341291 [D][MfUltralightPoller] Reading tearing flag 2
26341294 [D][MfUltralightPoller] No limits in authentication. Trying default password
26341302 [D][Nfc] FWT Timeout
26341304 [D][MfUltralightPoller] Read success
26341305 [D][MfUltralightPoller] Check writing capability
26341308 [D][MfUltralightPoller] Unknown password
26341310 [D][Nfc] FWT Timeout
26348114 [D][ViewDispatcher] View changed while key press 2000BBF8 -> 2000E218. Sending key: Back, type: Release, sequence: 00000180 to previous view port
```
Can't write to the card either.

**Comment by gornekich** [comment]
Hello! We reproduces the issue. The fix will arive with #3285 
It would be very helpful if you confirm that mentioned PR fix your issue

**Comment by johng3587** [comment]
Built and tested, the PR does fix the issue. Nice job!
